Services/DataX.Config/DataX.Config/ConfigGeneration/Processor/S450_PrepareTransformFile.cs

```diff
@@ -51,26 +54,25 @@ public override async Task<string> Process(FlowDeploymentSession flowToDeploy)
             }
             string queries = string.Join("\n", guiConfig.Process?.Queries);
 
-            string ruleDefinitions =  RuleDefinitionGenerator.GenerateRuleDefinitions(guiConfig.Rules, config.Name);
+            string ruleDefinitions = RuleDefinitionGenerator.GenerateRuleDefinitions(guiConfig.Rules, config.Name);
             RulesCode rulesCode = CodeGen.GenerateCode(queries, ruleDefinitions, config.Name);
 
             Ensure.NotNull(rulesCode, "rulesCode");
 
             // Save the rulesCode object for downstream processing
             flowToDeploy.SetAttachment(AttachmentName_CodeGenObject, rulesCode);
 
-            var runtimeConfigBaseFolder = flowToDeploy.GetTokenString(PrepareJobConfigVariables.TokenName_RuntimeConfigFolder);
-            Ensure.NotNull(runtimeConfigBaseFolder, "runtimeConfigBaseFolder");
-
             var runtimeKeyVaultName = flowToDeploy.GetTokenString(PortConfigurationSettings.TokenName_RuntimeKeyVaultName);
             Ensure.NotNull(runtimeKeyVaultName, "runtimeKeyVaultName");
 
-            var filePath = ResourcePathUtil.Combine(runtimeConfigBaseFolder, $"{config.Name}-combined.txt");
-            var transformFilePath = await RuntimeStorage.SaveFile(filePath, rulesCode.Code);
             var secretName = $"{config.Name}-transform";
-            var transformFileSecret = await KeyVaultClient.SaveSecretAsync(runtimeKeyVaultName, secretName, transformFilePath, Configuration.TryGet(Constants.ConfigSettingName_SparkType, out string sparkType) ? sparkType : null);
+            Configuration.TryGet(Constants.ConfigSettingName_SparkType, out string sparkType);
+            var uriPrefix = (sparkType != null && sparkType == Constants.SparkTypeDataBricks) ? Constants.PrefixSecretScope : Constants.PrefixKeyVault;
```

Comment:
This logic to compute uriPrefix is now in multiple places. Can we consolidate by adding a helper method to KeyVaultClient and call it from everywhere its needed including in KeyVaultClient. Only KeyVaultClient should know the nuances of uri based on sparkType.

Reply:
Makes sense. Created a helper and updated to use it.